### ISSUE TITLE
Functions Application Insights client does not pick up Request-Id HTTP Header

### ISSUE THREAD

**Issue by murdockcrc** [Azure/Azure-Functions#1100]
**Issue Description**
I am trying to pass in a client-generated OperationId to Azure Function execution calls on HTTP triggers. However, the value of this HTTP header is not passed through to the AI telemetry information as `OperationId`. I have tried passing in the following headers with custom values, but none of them seem to propagate the OperationId that is generated by the caller:

- `Request-Id`
- `x-ms-root-id`
- `x-ms-request-id`

**Expected Result**
When the client calls an Azure Function on an HTTP trigger with the HTTP header `Request-Id` in the request, the Functions runtime will propagate the value of that header as the `OperationId` in the Application Insights Request event.

**Comment by brettsam** [comment]
This is currently not supported, but is on our feature list. The broad list of features around distributed tracing is here, and Request-Id is on that list: https://github.com/Azure/azure-webjobs-sdk/issues/1731. 

Going to close this as we're tracking it elsewhere already.

@lmolkova FYI.

**Comment by alexkarcher-msft** [comment]
The workaround, for the time being is to log that data as a custom event in your application code. Then you will be able to write a custom query to aggregate the data in App Insights log analytics.
https://docs.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics

**Comment by tslavik** [comment]
@alexkarcher-msft yes this could be workaround if your function is last in the chain. But for example when you store something to DB you cannot see it in E2E transaction and also in application map. 
@brettsam I have frontend written in angular and call through API Management Java backend with cosmosdb - correlation works great (in cosmos only HTTP tracked). Then I add Azure function in c# that store something to sql server and it is called from Java backend. Am I understand correctly that now is not possible to correlate following scenario to see it in E2E transaction?
